api/README.md

```diff
@@ -1 +1,269 @@
+- `PATCH /posts/31/comments/25`   - Partially updates the comment with id 25 on post with id 31.
+- `DELETE /posts/31/comments/25`  - Deletes the comment with id 25 on post with id 31.
+
+If a resource can exist independently it's advised to just include the resource's identifier in the parent. Alternatively if the client needs multiple related resources at the same time they should be bundled together into the same response to reduce the number of requests.
+
+### Actions
+
+After the resources are defined you need to find out what kind of actions should be applied to them. The basic CRUD actions can be omitted by simply using the HTTP request methods mentioned above. If an action doesn't fit into the CRUD operations there are two main approaches you can choose from:
+
+1. Treat it like a sub-resource and modify it with CRUD operations. E.g. `PUT /posts/31/like` to like the post and `DELETE /posts/31/like` to dislike the post.
+1. There are cases when you can't map the action to a single resource. In this case naming the endpoint with the name of the action makes the most sense, e.g. `/search`, `/sign-in`, etc.
+
+**You should never ever** leak any implementation details via your API.
+
+## HTTP Request Methods
+
+The HTTP request methods `GET`, `POST`, `PUT`, `PATCH` and `DELETE` have specific meaning in manipulating these.
+
+### GET
+
+- Retrieves a resource or a list of resources.
+- It should **never** update a resource.
+- Shouldn't even update the access logs.
+- Use query strings instead of request body.
+
+To retrieve the list of posts filtered by the keyword "awesome":
+```HTTP
+GET /posts?search=awesome
+```
+
+To retrieve the post with id 31:
+```HTTP
+GET /posts/31
+```
+
+### POST
+
+- Creates a resource.
+- It should be used when the server has to generate a unique key for the created resource.
+- Multiple POST requests to the same URL create duplicate records.
+
+To create a post:
+```HTTP
+POST /posts
+Request payload:
+{
+  "title": "Awesome title",
+  "content": "even more awesomeness"
+}
+```
+
+> The newly created resource (with a unique identifier) should be returned in the response body.
+
+### PUT
+
+- Creates/updates a resource.
+- Multiple PUT requests to the same URL don't create/update duplicate records.
+- It should be used when the client is aware of the url of the resource.
+
+To update the post with id 31:
+```HTTP
+PUT /posts/31
+Request payload:
+{
+  "title": "Awesome title",
+  "content": "even more awesomeness"
+}
+```
+
+> If creating/updating the resource doesn't trigger unpredicted changes in the underlaying data there is no need to return the resource in the response body.
+
+### PATCH
+
+- Partially updates a resource.
+- Updated attributes of the resource should be provided only.
+
+To update the content of the post with id 31:
+```HTTP
+PATCH /posts/31
+Request payload:
+{
+  "content": "less awesomeness"
+}
+```
+
+> No response body should be returned.
```

Comment:
I am not sure we can handle no responses that well. The library we are using expects a JSON response, and fails serializing nil responses. We ran into this on a few occasions with SmartUp API.
I will do some research if we can somehow solve this easily on client-side. If not, we should stabilize a valid JSON response for these cases